src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      let element = this[i];
+
+      if (this[i] === null || this[i] === undefined) {
```

Comment:
Replace with `!this[i]`, Its the same as `this[i] === null || this[i] === undefined`

Reply:
It's not the same in this case, if replaced, the function fails the tests because the loop ignores 0, and NaN.